internal/util-collection/src/main/scala/sbt/internal/util/Settings.scala

```diff
@@ -35,10 +35,10 @@ private final class Settings0[Scope](
     data.flatMap { case (scope, map) => map.keys.map(k => f(scope, k)) }.toSeq
 
   def get[T](scope: Scope, key: AttributeKey[T]): Option[T] =
-    delegates(scope).toStream.flatMap(sc => getDirect(sc, key)).headOption
+    delegates(scope).flatMap(sc => getDirect(sc, key)).headOption
 
   def definingScope(scope: Scope, key: AttributeKey[_]): Option[Scope] =
-    delegates(scope).toStream.find(sc => getDirect(sc, key).isDefined)
+    delegates(scope).find(sc => getDirect(sc, key).isDefined)
```

Comment:
do they help memory consumption though, by being lazy?

Reply:
Assuming find returns at first hit, I am not sure if this saves either time or space.

Reply:
If we had an iterator version, as suggested by Jason below, it would be a different story. But I'd say that would be another PR.

Reply:
Probably related https://docs.scala-lang.org/tutorials/FAQ/stream-view-iterator.html

Reply:
Delegation chain shouldn't be too long (< 100 elements usually) and the objects are short-lived, so without a bteter idea like the iterator, I'd also say, the extra memory usage doesn't matter here.

For `find` the big problem was the aggregation caching because in most cases `aggregate in X` is not defined but on the end of the delegation chain in which case using `Stream` was super expensive.